src/cpp/core/system/Xdg.cpp

```diff
@@ -108,17 +109,53 @@ FilePath resolveXdgDir(const std::string& envVar,
 } // anonymous namespace
 
 FilePath userConfigDir(
-        const boost::optional<std::string>& user,
-        const boost::optional<FilePath>& homeDir)
+   const boost::optional<std::string>& user,
+   const boost::optional<FilePath>& homeDir)
 {
-   return resolveXdgDir("XDG_CONFIG_HOME", 
+   FilePath confDir = resolveXdgDir("XDG_CONFIG_HOME", 
 #ifdef _WIN32
          FOLDERID_RoamingAppData,
 #endif
          "~/.config",
          user,
          homeDir
    );
+
+   Error permError;
+   Error error = confDir.ensureDirectory();
+   if (!error)
+   {
+      // Test permissions here.
+      error = confDir.testWritePermissions();
+      if (error)
+      {
+         permError = Error(
+            error.getName(),
+            error.getCode(),
+            "Missing write permissions to " + confDir.getAbsolutePath() + ". Some features may not work correctly.",
+            ERROR_LOCATION);
+
+         system::User currUser;
+         error = system::User::getUserFromIdentifier(system::username(), currUser);
+         if (!error)
+            error = confDir.changeOwnership(currUser, true);
+      }
+   }
+
+   if (error)
+   {
+      std::string message;
+      message.append("Encounterd an issue while testing config dir \"")
```

Comment:
```suggestion
      message.append("Encountered an issue while testing config dir \"")
```

Reply:
I changed the phrasing of this locally to `"Unable to access the directory \""`.